src/server/acceptor.rs

```diff
@@ -18,9 +18,8 @@ impl<A: NetworkAcceptor> AcceptorPool<A> {
     /// ## Panics
     ///
     /// Panics if threads == 0.
-    pub fn accept<F: Fn(A::Stream) + Send + Sync>(self,
-                                                  work: F,
-                                                  threads: usize) -> JoinGuard<'static, ()> {
+    pub fn accept<F>(self, work: F, threads: usize) -> JoinGuard<'static, ()>
+        where F: Fn(A::Stream) + Send + Sync + 'static {
```

Comment:
Additional advantage of making this blocking: `'a` here instead of `'static`, so this can be used with non-'static data.